docs/dependencies.md

```diff
@@ -1,80 +1,196 @@
+1. Dependencies are always in sync and are managed by Particle
+
+### Building Particle firmware
+
+```bash
+docker run --rm --volume <local firmware repository>:/particle-iot/firmware --workdir /particle-iot/firmware/modules particle/toolchain clean all PLATFORM=(core|electron|p1|photon)
+```
+
+**Option breakdown:**
+
+- `<local firmware repository>` - This is the path to the firmware folder on your local machine.
+- `(core|electron|p1|photon)` - Select the option from this list that matches the device for which you are compiling.
+
+### Building a user application
+
+```bash
+docker run --rm --volume <local firmware repository>:/particle-iot/firmware --volume <user application folder>:/particle-iot/app --workdir /particle-iot/firmware/main particle/toolchain clean all PLATFORM=(core|electron|p1|photon) APPDIR=/particle-iot/app TARGET_FILE=<user application>
+```
+
+**Option breakdown:**
+
+- `<local firmware repository>` - This is the path to the firmware folder on your local machine.
+- `<user application folder>` - This is the path to the user application folder on your local machine.
+- `(core|electron|p1|photon)` - Select the option from this list that matches the device for which you are compiling.
+- `<user application>` - This will be the name of the resulting binary (i.e. `my_app.elf`).
+
+> _**NOTE:** Special thanks to Kevin Sidwar for pioneering containerized builds in [his blog](https://www.kevinsidwar.com/iot/2018/3/1/local-particle-build-and-debug-through-docker-on-mac)._
+
+## Native Dependency Installation
+
+### Dependency List
+
+- [`arm-none-eabi-gcc`](#gcc-for-arm-cortex-processors)
+- [`make`](#gnu-make)
+- [`crc32`](#32-bit-cyclic-redundency-check-utility)
+- [`xxd`](#xxd-hex-dump-utility)
+- [Zadig (Windows only)](#zadig-windows-only)
+
+### GCC for ARM Cortex processors
+
+| Confirmed version | Download page |
+|:-----------------:|:-------------:|
+| 5.3.1 | [launchpad.net](https://launchpad.net/gcc-arm-embedded) |
 
-#### 1. GCC for ARM Cortex processors
 The Core/Photon uses an ARM Cortex M3 CPU based microcontroller. All of the code is built around the GNU GCC toolchain offered and maintained by ARM.
 
 The build requires version 5.3.1 20160307 or newer of ARM GCC and will print an error
 message if the version is older than this.
 
+| OS | Distribution | Package Manager | Command |
+|:-- |:------------ |:--------------- |:------- |
+| Linux | Ubuntu | apt-get | `sudo add-apt-repository ppa:team-gcc-arm-embedded/ppa && sudo apt-get update && sudo apt-get install gcc-arm-embedded` |
+| Linux | _(non-Ubuntu)_ | n/a | `# See detailed instructions below` |
+| OSX | n/a | [Homebrew](http://brew.sh/) | `# See detailed instructions below` |
+| Windows | n/a | n/a | `# See detailed instructions below` |
+
+**Linux _(non-Ubuntu)_:**
+
+Install from the download page.
 
-**Linux and Windows**:
-- Download and install version 5.3.1 from: https://launchpad.net/gcc-arm-embedded
+Additional 32-bit dependencies:
```

Comment:
I think we should generalize this section and remove Debian-specific references.